### ISSUE TITLE
Change textstat_frequency column name for weighted dfm

### ISSUE THREAD

**Issue by stefan-mueller** [kbenoit/quanteda#914]
I mentioned the problem under #897 already, but I thought I create a separate issue.

So far, `texstat_frequency` does not distinguish between weighted and non-weighted dfm's. In all cases, the column is named `frequency` even if `texstat_frequency` is applied to a weighted dfm.

```r
obamadfm <- dfm(data_corpus_inaugural[56:57], 
                remove_punct = TRUE, remove = stopwords("english"))

head(textstat_frequency(obamadfm), 2)

# feature frequency rank docfreq
# 1      us        44    1       2
# 2    must        25    2       2

obamadfm_weight <- dfm_weight(obamadfm, type = "relfreq")

head(textstat_frequency(obamadfm_weight), 2)

# feature  frequency rank docfreq
# 1      us 0.03987811    1       2
# 2    must 0.02322939    2       2
```

I thought it might be good to rename the `frequency` column based on the `type` argument from `dfm_weight` (which can be "frequency", "relfreq", "relmaxfreq", "logfreq",  "tfidf"). This makes it easy to see which type of frequency was estimated and enables us to assign the correct x-axis label in the new `textplot_frequency` function.

**Comment by kbenoit** [comment]
I'm not sure it makes sense to sum non-count weights, other than possibly "relfreq". It would certainly not make sense to sum "logfreq". I'm wondering under what meaningful use case we would want to support the weights at all. Ideas?

**Comment by stefan-mueller** [comment]
I think "tfidf" could make sense (of course only if `textstat_frequency` is calculated for different groups). Silge and Robinson mainly compare the tfidf in the chapter you sent me a while ago (http://tidytextmining.com/tfidf.html). And I think it could be nice to plot the inverse document frequency to get a sense of importance of a word in one group compared to the other groups. Correct me if I'm wrong or if this is not the intention of `textstat_frequency`.

**Comment by kbenoit** [comment]
You're right, it is possible to perform this weighting, but only after the grouping has occurred. So the valid way would be:

```r
dfm_group(x, groups) %>% dfm_weight(type = "whatever") %>% textstat_frequency()
```
or , add an argument to textstat_frequency to be fed to `dfm_weight()` through `...`:
```r
textstat_frequency(x, groups, ...)
```
e.g.
```r
textstat_frequency(presdfm, groups = "President", type = "tfidf")
```

**Comment by stefan-mueller** [comment]
Yes, this is exactly what I thought. Adding a `type` argument to `textstat_frequency` would probably be the best way to do this. In my former PR ( #900) I basically added the example you have as your first option (and therefore used a `dfm` object as the input). But I agree with you that `textplot_frequency` should be based on `textstat_frequency`. Therefore, it would be great to have a `type` argument in `textstat_frequency`. Could you maybe implement this? Afterwards, I'm happy to make a new attempt for `textplot_frequency`.

**Comment by kbenoit** [comment]
I'm closing this issue since I don't think it makes sense to aggregate weighted document frequencies. The better approach is the group, weight, textstat_frequency as above.
